openwisp_monitoring/device/tests/mock.py

```diff
@@ -0,0 +1,536 @@
+
+
+# Work in Progress
+def mock_get_csv(request, pk):
+    r = HttpResponse(mock_csv_data, content_type='text/csv', status=200)
+    r['Content-Disposition'] = 'attachment; filename=data.csv'
+    return r
+
+
+def mock_metric_view(request, data, pk):
+    statistics_stored = ['rx_bytes', 'tx_bytes']
+    ct = ContentType.objects.get(model='device', app_label='config')
+    for interface in data.get('interfaces', []):
+        ifname = interface['name']
+        for key, value in interface.get('statistics', {}).items():
+            if key not in statistics_stored:
+                continue
+            name = '{0} {1}'.format(ifname, key)
+            metric, created = Metric._get_or_create(object_id=pk,
+                                                    content_type=ct,
+                                                    key=ifname,
+                                                    field_name=key,
+                                                    name=name)
+            if created:
+                if not (metric.field_name != 'tx_bytes' or 'traffic' not in monitoring_settings.AUTO_GRAPHS):
+                    graph = Graph(metric=metric,
+                                  description=('{0} traffic (GB)').format(metric.key),
+                                  query="SELECT SUM(tx_bytes) / 1000000000 AS upload, "
+                                        "SUM(rx_bytes) / 1000000000 AS download FROM {key} "
+                                        "WHERE time >= '{time}' AND content_type = '{content_type}' "
+                                        "AND object_id = '{object_id}' GROUP BY time(24h) fill(0)")
+                    graph.full_clean()
+                    graph.save()
+        try:
+            clients = interface['wireless']['clients']
+        except KeyError:
+            continue
+        name = '{0} wifi clients'.format(ifname)
+        metric, created = Metric._get_or_create(object_id=pk,
+                                                content_type=ct,
+                                                key=ifname,
+                                                field_name='clients',
+                                                name=name)
+        for client in clients:
+            if 'mac' not in client:
+                # print('Here!')
+                continue
+            metric.write(client['mac'])
+        if created:
+            if not 'wifi_clients' not in monitoring_settings.AUTO_GRAPHS:
+                graph = Graph(metric=metric,
+                              description=('{0} clients').format(metric.key),
+                              query="SELECT COUNT(DISTINCT({field_name})) AS value FROM {key} "
+                                    "WHERE time >= '{time}' AND content_type = '{content_type}' "
+                                    "AND object_id = '{object_id}' GROUP BY time(24h) fill(0)")
+                graph.full_clean()
+                graph.save()
+    return Response(status=200)
```

Comment:
You either make the main code callable and use the main code or you don't do anything, you can't duplicate the logic of the views because that adds maintenance overhead.